pyDeltaRCM/deltaRCM_tools.py

```diff
@@ -445,3 +449,40 @@ def save_grids(self, var_name, var, ts):
         except:
             self.logger.error('Cannot save grid to netCDF file.')
             warnings.warn(UserWarning('Cannot save grid to netCDF file.'))
+
+    def save_checkpoint(self):
+        """Save checkpoint files.
+
+        Saves the grids to a .npz file so that the model can be
+        initiated from this point. The timestep of the checkpoint is also
+        saved. The values from the model that are saved to the checkpoint.npz
+        are the following:
+
+        - Model time
+        - Flow velocity and its components
+        - Water depth
+        - Water stage
+        - Topography
+        - Current random seed state
+
+        Stratigraphy is held as sparse matrices, so each constituent sparse
+        matrix is saved in its own .npz file.
+
+        - Stratigraphic 'topography' in 'strata_eta.npz'
+        - Stratigraphic sand fraction in 'strata_sand_frac.npz'
+
+        Note, checkpoints are re-written everytime the save iterval is
+        reached if the checkpointing flag is set to 'True'.
+
+        """
+        ckp_file = os.path.join(self.prefix, 'checkpoint.npz')
+        ckp_time = self._time + 1  # advance time for the checkpoint
+        np.savez_compressed(ckp_file, time=ckp_time, uw=self.uw, ux=self.ux,
+                            uy=self.uy, depth=self.depth, stage=self.stage,
+                            eta=self.eta, rng_state=np.random.get_state())
+
+        ckp_strata_eta = os.path.join(self.prefix, 'strata_eta.npz')
+        save_npz(ckp_strata_eta, self.strata_eta.tocsc())
+
+        ckp_sand_frac = os.path.join(self.prefix, 'strata_sand_frac.npz')
+        save_npz(ckp_sand_frac, self.strata_sand_frac.tocsc())
```

Comment:
So these cannot go into the same `checkpoint.npz` file above, because they are sparse arrays? Can they go into the same file with one another?

I guess it's fine however it works, but maybe you can name the files more descriptively. Consider something like:
* `checkpoint.npz` --> `checkpoint_grids.npz`
* `strata_eta.npz` --> `checkpoint_strata_eta.npz`
* `strata_sand_frac.npz` --> `checkpoint_strata_sand_frac.npz`

Reply:
Found a solution to this ([woohoo](https://stackoverflow.com/questions/8955448/save-load-scipy-sparse-csr-matrix-in-portable-data-format)) and now they are all in a single `checkpoint.npz` file.